mockgen/internal/tests/generics/source/assert_test.go

```diff
@@ -0,0 +1,12 @@
+package source
+
+import (
+	"testing"
+
+	"go.uber.org/mock/mockgen/internal/tests/generics"
+)
+
+func TestAssert(t *testing.T) {
+	var x MockEmbeddingIface[int, float64]
+	var _ generics.EmbeddingIface[int, float64] = &x
```

Comment:
Should we add assertions for the new test interface types as well? I see that Earth is embedded into EmbeddingIface, but Universe isn't overlapping.

Should we also call the Mock constructor instead of instantiating the mock type directly? I think this would be a more realistic example of how the user would construct a mock.